# Bioinformatics tools


**Primer3**: Primers design (http://bioinfo.ut.ee/primer3-0.4.0/)

Primer check-ins: 
1.	**Blat** for non-specific annealing (if primers length â‰¥20 nt) (http://genome.ucsc.edu/cgi-bin/hgBlat)
2.	**IDT OligoAnalyzer** for self-dimers, hetero-dimers, hairpin, etc. (https://eu.idtdna.com/calc/analyzer)
3.	**USCS** to check for LINEs, SINEs, etc. https://genome.ucsc.edu/

Gene sequences search:   
 - **USCS**: Genome browser (generally ideal for DNA sequence) https://genome.ucsc.edu/   
 - **VEGA browser**: cDNA sequence, protein, domains, etc. http://vega.archive.ensembl.org/index.html

**The Human Protein Atlas**: Protein expression https://www.proteinatlas.org/   

**Firebrowser**: Gene expression http://firebrowse.org/   

**ExAC Browser Beta**: Population frequencies among other things http://exac.broadinstitute.org/   

**UCSC**: For restriction enzymes cut-sites

**GeneCards**: For genes information http://www.genecards.org/

Sequences analysis:
-	Manual analysis with **Chromas lite** software- free downloads from: https://technelysium.com.au/wp/chromas/
-	Automated analysis with **Mutation surveyor** software- license fee needed
-	**USCS-Blat** tool for sequence annealing (specially for homozygosis mutations): https://genome.ucsc.edu/cgi-bin/hgBlat

Mutations databases:
-	**HGMD**: germline mutations  http://www.hgmd.cf.ac.uk/ac/index.php
-	**LOVD-INSIGHT**: especially for MMR genes variants https://www.insight-group.org/variants/databases/
-	**COSMIC**: Somatic mutations http://cancer.sanger.ac.uk/cosmic
-	**IARC TP53 database**: TP53 mutations http://p53.iarc.fr
-	**The P53 Web Site**: http://p53.fr
-	**Mismatch repair Gene Variant Database**: www.med.mun.ca/mmvariants/default.aspx
-	**miRBASE database**: sequences  of all miRNAs published so far and including miRBAse Registry and microcosm to predict miRNA targets in a given sequence www.mirbase.org

Mutation in silico studies:

-	**PolyPhen-2** http://genetics.bwh.harvard.edu/pph2/
-	**SIFT** http://sift.jcvi.org/
-	**Pmut** http://mmb.irbbarcelona.org/PMut/
-	**Panther** http://pantherdb.org/
-	**Align-GVGD** http://agvgd.iarc.fr/
-	**MicroRNA**: study of target sequences to which miRNAs bind www.microrna.org/
-	**ESEFinder 3.0**: study of variants that can affect to SR binding domains and therefore to the binding of these proteins to ESE sites http://krainer01.cshl.edu/cgi-bin/tools/ESE3/esefinder.cgi?process=home
-	**RESCUE_ESE**: for detection of sequences with ESE activity http://genes.mit.edu/burgelab/rescue-ese/
-	**TFSEARCH 1.3**: detection of promoter sequences to which known transcription factors bind http://diyhpl.us/~bryan/irc/protocol-online/protocol-cache/TFSEARCH.html
-	**PSIPRED**: program for the study of structural modifications generated by VUS http://bioinf.cs.ucl.ac.uk/psipred/

VCF
- **JVARKIT**: Java utilities for Bioinformatics http://lindenb.github.io/jvarkit/
